### DISCUSSION TITLE
Why do you need to increase stack pointer by 2.

### DISCUSSION THREAD

**Discussion by NeeEoo** [elikaski/BF-it#60] [Q&A]
If the code has 1 extra cell like in if else. You need to increment the stack pointer by 2. Why can't you increment the stack pointer by 1.

If it doesn't have an extra cell there isn't any call to increase stack pointer. Such as in `compile_while`

**Comment by elikaski** [comment]
1 cell for evaluating the expression
1 cell for "do we need to execute else?"
Similar to the switch-case statement
I tried to think of a way to implement it with just one cell but I didn't find one

**Reply by NeeEoo** [reply]
I think i understand it a little better.

In `compile_if` since the expression cell isn't used again after wouldn't it be possible to rearrange it so that the expression cell appears in the temp cells?
